src/py/flwr/cli/install.py

```diff
@@ -0,0 +1,208 @@
+
+
+def validate_and_install(
+    project_dir: Path,
+    fab_name: str,
+    flwr_dir: Optional[Path],
+    skip_prompt: bool = False,
+) -> None:
+    """Validate TOML files and install the project to the desired directory."""
+    config, _, _ = load_and_validate(project_dir / "pyproject.toml", check_module=False)
+
+    if config is None:
+        typer.secho(
+            "❌ Invalid config inside FAB file.",
+            fg=typer.colors.RED,
+            bold=True,
+        )
+        raise typer.Exit(code=1)
+
+    username = config["flower"]["publisher"]
+    project_name = config["project"]["name"]
+    version = config["project"]["version"]
+
+    if fab_name != f"{username}.{project_name}.{version.replace('.', '-')}":
+        typer.secho(
+            "❌ FAB file has incorrect name, it should be "
+            "`<username>.<project_name>.<version>.fab`.",
+            fg=typer.colors.RED,
+            bold=True,
+        )
+        raise typer.Exit(code=1)
+
+    install_dir: Path = (
+        (
+            Path(
+                os.getenv(
+                    "FLWR_HOME",
+                    f"{os.getenv('XDG_DATA_HOME', os.getenv('HOME'))}/.flwr",
+                )
+            )
+            if not flwr_dir
+            else flwr_dir
+        )
+        / "apps"
+        / username
+        / project_name
+        / version
+    )
+    if install_dir.exists() and not skip_prompt:
+        if not typer.confirm(
+            typer.style(
+                f"\n💬 {project_name} version {version} is already installed, "
+                "do you want to reinstall it?",
+                fg=typer.colors.MAGENTA,
+                bold=True,
+            )
+        ):
+            return
+
+    install_dir.mkdir(parents=True, exist_ok=True)
+
+    # Move contents from source directory
+    for item in project_dir.iterdir():
+        if item.is_dir():
+            shutil.copytree(item, install_dir / item.name, dirs_exist_ok=True)
+        else:
+            shutil.copy2(item, install_dir / item.name)
+
+    typer.secho(
+        f"🎊 Successfully installed {project_name} to {install_dir}.",
+        fg=typer.colors.GREEN,
+        bold=True,
+    )
+
+
+def _verify_hashes(list_content: str, tmpdir: Path) -> bool:
+    """Verify file hashes based on the LIST content."""
+    for line in list_content.strip().split("\n"):
+        rel_path, hash_expected, _ = line.split(",")
+        file_path = tmpdir / rel_path
+        if not file_path.exists() or _get_sha256_hash(file_path) != hash_expected:
+            return False
+    return True
+
+
+def _get_sha256_hash(file_path: Path) -> str:
+    """Calculate the SHA-256 hash of a file."""
+    sha256 = hashlib.sha256()
+    with open(file_path, "rb") as f:
+        while True:
+            data = f.read(65536)
+            if not data:
+                break
+            sha256.update(data)
+    return sha256.hexdigest()
```

Comment:
We have a similar function in `flwr.cli.build`, can we merge them? We can move it to `utils`